src/fragments/start/getting-started/reactnative/auth.mdx

```diff
@@ -0,0 +1,263 @@
+
+## Create authentication service
+
+To add authentication to your app, run this command:
+
+```bash
+amplify add auth
+```
+
+Select the defaults for the following prompts:
+
+```console
+? Do you want to use the default authentication and security configuration? Default configuration
+Warning: you will not be able to edit these selections. 
+? How do you want users to be able to sign in? Username
+? Do you want to configure advanced settings? No, I am done.
+```
+
+To deploy the service, run the `push` command:
+
+```bash
+amplify push
+```
+
+Now, the authentication service has been deployed and you can start using it. To view the deployed services in your project at any time, go to Amplify Console by running the following command:
+
+```bash
+amplify console
+```
+
+## Create login UI
+
+Creating a login flow can be quite difficult and time consuming to get right. Luckily, Amplify UI has an [authentication](https://ui.docs.amplify.aws/react-native/connected-components/authenticator) component that provides an entire authentication flow for you, using the configuration you specified in **aws-exports.js.**
+
+### Install Amplify UI
+
+The `@aws-amplify/ui-react-native` package includes React Native specific UI components you'll use to build your app. Install it with the following command:
+
+<BlockSwitcher>
+<Block name="Expo CLI">
+
+```bash
+npm install @aws-amplify/ui-react-native react-native-safe-area-context react-native-get-random-values react-native-url-polyfill
+```
+
+</Block>
+<Block name="React Native CLI">
+
+```bash
+yarn add @aws-amplify/ui-react-native react-native-safe-area-context react-native-get-random-values react-native-url-polyfill
+```
+
+You will also need to install the pod dependencies for iOS:
+
+```sh
+npx pod-install
+```
+
+</Block>
+</BlockSwitcher>
+
+### Add the Amplify UI Authenticator component
+
+<BlockSwitcher>
+<Block name="Expo CLI">
+
+Open **App.js** and make the following changes:
+
+</Block>
+<Block name="React Native CLI">
+
+Open **App.tsx** and make the following changes:
+
+</Block>
+</BlockSwitcher>
+
+1. Import the `withAuthenticator` Higher-Order Component and `useAuthenticator` hook:
+
+```javascript
+
+import { withAuthenticator, useAuthenticator } from '@aws-amplify/ui-react-native';
+
+````
+
+2. The `useAuthenticator` hook can be added to any component that needs it. Create a custom `SignOutButton` component that will be used within the App component, and destructure the `{ user, signOut }` properties returned in the context.
+
+```javascript
+
+const SignOutButton = () => {
+  const { user, signOut } = useAuthenticator((context) => [context.user]);
```

Comment:
Might be helpful for users to explain the use of the selector param passed to `useAuthenticator`

Reply:
added detailed explanation about the hook and callback